ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -252,16 +255,42 @@ pub fn process_withdrawals<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _execution_payload: &ExecutionPayload<
+    execution_payload: &ExecutionPayload<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
         MAX_BYTES_PER_TRANSACTION,
         MAX_TRANSACTIONS_PER_PAYLOAD,
         MAX_WITHDRAWALS_PER_PAYLOAD,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let expected_withdrawals = get_expected_withdrawals(state, context).unwrap();
+    assert_eq!(execution_payload.withdrawals.len(), expected_withdrawals.len());
+    for (expected_withdrawal, withdrawal) in
+        expected_withdrawals.iter().zip(execution_payload.withdrawals.iter())
+    {
+        assert_eq!(withdrawal, expected_withdrawal);
+        decrease_balance(state, withdrawal.validator_index, withdrawal.amount);
+    }
```

Comment:
we can rely on the implementation of `PartialEq` here to do a lot of this for us... also, we want to return an error, rather than just assert something (as it is not nice to use this as a library and just have it panic on you)

```suggestion
    if execution_payload.withdrawals != expected_withdrawals {
        return Err(..);
    }
    for withdrawal in &expected_withdrawals {
        decrease_balance(state, withdrawal.validator_index, withdrawal.amount);
    }
```

Reply:
I'd define an error variant for `Withdrawal`s like the others here: https://github.com/ralexstokes/ethereum-consensus/blob/61836a64464f311a40fd3038edf774f5c239e3ea/ethereum-consensus/src/state_transition/error.rs#L68